cmd/common.go

```diff
@@ -0,0 +1,14 @@
+package cmd
+
+import "os"
+
+func resolveEnvironment(args []string, defaultEnvironment string) string {
+	environment := defaultEnvironment
```

Comment:
I think the defaultEnvironment should be a constant for the project, I don't see why it should be specific to the caller (meaning different part of the app may do different env assumptions at the same time).

And one of the goal of using env variable, it is not have to pass them by arg with all the "complex" machinery of passing them everywhere. If the visibility of env var is problematic through the Makefile, you can call it directly with the makefile  `ALGOREA_ENV=test make something` or `make something ALGOREA_ENV=test`

So I would remove most of the code of this commit (sorry 😕 but it is just a few lines 🙂). But maybe I would add a check that it is among "prod", "test, "dev", and otherwise default to "dev" (or other default if you prefer). Then you could add shortcut function "inProdEnv() bool", "inTestEnv()", ...